### PR TITLE
libxc: add v7.0.0

### PR DESCRIPTION
**Pull request by prckent**
Released 2024-10-09.

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
Confirmed the version sha256.

**Comment by tldahlgren** [comment]
Looks like the new version breaks the build of `cp2k`.

**Comment by prckent** [comment]
@tldahlgren Are we waiting on CP2K to make fixes or is this going to be merged first? It is not clear to me which codes are considered a hard requirement to build, or how a code gets on that list. The CP2K GitHub reports some success and some problems with libxc 7.0, e.g. https://github.com/cp2k/cp2k/issues/3724 . If we are waiting for an update, it fortunately does not seem like a major issue.

**Comment by tldahlgren** [comment]
> @tldahlgren Are we waiting on CP2K to make fixes or is this going to be merged first? It is not clear to me which codes are considered a hard requirement to build, or how a code gets on that list. The CP2K GitHub reports some success and some problems with libxc 7.0, e.g. [cp2k/cp2k#3724](https://github.com/cp2k/cp2k/issues/3724) . If we are waiting for an update, it fortunately does not seem like a major issue.

CI checks are required to pass in order for a PR to be merged.

**Comment by lang-m** [comment]
@tldahlgren Out of curiosity: why is cp2k part of the libxc pipelines? Libxc is used by many (if not most) big DFT/quantum chemistry codes, several of which are likely not (yet) compatible with libxc 7. So why cp2k specifically and not any other DFT code (libxc and cp2k seem to be completely independent, no overlap in maintainers or similar)?

File: var/spack/repos/builtin/packages/libxc/package.py
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/libxc/package.py:19]: @prckent, @lang-m, here is a suggestion that might fix CI by not making 7.0.0 the preferred version yet but instead keeping 6.2.2 as the preferred version:
```suggestion
    version(
        "6.2.2",
        sha256="d1b65ef74615a1e539d87a0e6662f04baf3a2316706b4e2e686da3193b26b20f")
        preferred=True,
    )
```
With this, recipes that need `libxc@7:` can already depend on it, and recipes can also be updated to declare which versions need to depend on `libxc@:6` can add those version ranges too.

This way of integrating libxc-7.0.0 allows testing all users of `libxc` with the new version (e.g. by forcing the build with `recipe ^libxc@7`, but the default builds (in CI and outside of CI) are not affected by adding libxc-7.0.0.

Comment (prckent) [reply var/spack/repos/builtin/packages/libxc/package.py:19]: Thanks for the suggestion, but I think the preferred version actually should be the latest, and pinning the version is not actually preferred. Based on the release notes (1) v7 includes fixes for physically correct results for some uses (2) only programs depending on certain parts of the Fortran API in certain ways will need an update. Packages using C and Python interfaces will likely be fine and need no update. In this case I think most packages will not need any update at all. Any that do can consider implementing their own policy that best suits them.

**Comment by tldahlgren** [comment]
> @tldahlgren Out of curiosity: why is cp2k part of the libxc pipelines? Libxc is used by many (if not most) big DFT/quantum chemistry codes, several of which are likely not (yet) compatible with libxc 7. So why cp2k specifically and not any other DFT code (libxc and cp2k seem to be completely independent, no overlap in maintainers or similar)?

The reason for now is that several E4S stacks are building  `cp2k` and the package's `libxc` variant is enabled by default.

Does that help?

P.S. One long-term goal of Spack to eventually build *every* modified package of a PR.

**Pull request merged**
